### PR TITLE
Refactoring v2

### PR DESCRIPTION
**Pull request by andreoav**
🚧 This is a working in progress 🚧

# Introduction

Replacement for #36 

This proposal keeps packages within a monorepo structure but uses https://github.com/palmerhq/tsdx with minor tweaks to handle builds.

- [x] Migrate all packages to the new structure
- [ ] Code optimization and refactoring
- [ ] Rebase master into this branch to get the latest updates
- [ ] Update documentation
- [ ] Restore existing contributors
- [ ] Generate a pre-release
- [ ] Automate the publishing process
- [ ] Final release

Migration status:

- [ ] format-boleto
- [ ] format-cnpj
- [ ] format-cpf
- [ ] format-processo-juridico
- [ ] generate-cnpj
- [ ] generate-cpf
- [x] helper-generate-checksum
- [ ] helper-is-last-char
- [x] helper-only-numbers
- [ ] is-valid-boleto
- [ ] is-valid-cep
- [x] is-valid-cnpj
- [x] is-valid-cpf
- [ ] is-valid-email
- [ ] is-valid-phone
- [ ] is-valid-pis

❗️ IMPORTANT: This PR **DO NOT RESET THE REPOSITORY HISTORY**, otherwise all commits from all contributors will be lost.

### CODE REVIEW

File: helpers/helper-only-numbers/package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "@brazilian-utils/helper-only-numbers",
+  "version": "1.0.0-alpha.1",
+  "author": "Hyan Mandian <hyanmandian@hotmail.com>",
+  "repository": "https://github.com/brazilian-utils/brazilian-utils/tree/master/packages/helper-only-numbers",
+  "bugs": {
+    "url": "https://github.com/brazilian-utils/brazilian-utils/issues"
+  },
+  "license": "MIT",
+  "main": "dist/index.js",
```
Comment (hyanmandian) [inline helpers/helper-only-numbers/package.json]: maybe missing umd:main?

File: helpers/helper-only-numbers/src/index.ts [hunk 2]
```diff
@@ -1,3 +1,3 @@
 export default function onlyNumbers(str: string | number) {
-  return String(str).replace(/[^\d]/g, '');
+  return str.toString().replace(/[^\d]/g, '');
```
Comment (hyanmandian) [inline helpers/helper-only-numbers/src/index.ts]: I think that `String(str)` is better because if you receive a number you can't do `.toString()`

Comment (andreoav) [reply helpers/helper-only-numbers/src/index.ts]: @hyanmandian `Number` prototype has a `toString` method btw. I also wrote a test that handle this scenario.

https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Number/toString

File: helpers/helper-only-numbers/package.json (hunk 1 above)
Comment (andreoav) [reply helpers/helper-only-numbers/package.json]: Is it worthy? UMD bundle are bigger and I think using bundlers like react or rollup are pretty standard today.

Look at this issue: https://github.com/palmerhq/tsdx/issues/120

File: helpers/helper-only-numbers/src/index.ts (hunk 2 above)
Comment (fxamauri) [reply helpers/helper-only-numbers/src/index.ts]: `String(str)` will behave better if `null` passed as a parameter, but we are using types. I do not know if this is a problem.

File: helpers/helper-only-numbers/package.json (hunk 1 above)
Comment (hyanmandian) [reply helpers/helper-only-numbers/package.json]: I agree with you, so maybe we should remove all umb bundle for our packages. What do you think about it?

Comment (andreoav) [reply helpers/helper-only-numbers/package.json]: Yep. This PR doesn't include a UMD bundle.

**Pull request closed without merging**